api/spec/requests/spree/api/checkouts_controller_spec.rb

```diff
@@ -59,8 +59,7 @@ module Spree
       end
 
       it "will return an error if the order cannot transition" do
-        skip "not sure if this test is valid"
-        order.bill_address = nil
+        order.ship_address = nil
```

Comment:
What about grouping the order updates in one single instruction so we hit the DB only once? Something like `order.update!(ship_address: nil, state: "address)`

Reply:
Done @spaghetticode . Also, added a _note_ to the pr!